src/raytracer/xmlreader.m

```diff
@@ -63,114 +65,149 @@
-        normal = cross(vector2,vector1) * (1-(2*IndoorSwitch));
-        normal = round(normal/norm(normal),4);
-        vector3 = v2;
-        % for box. remove for others
-        D = -dot(normal,vector3);
-        
-        % Storing Material information in output if the material exists in the material database
-        if materialSwitch==1
-            switch2=0;
-            for iterateMaterials=1:size(MaterialLibrary,1)
-                if strcmpi(MaterialLibrary.Reflector{iterateMaterials},Material)
-                    CADOutput(indexCADOutput,14) = MaterialLibrary.PrimaryKey(iterateMaterials);
-                    switch2=1;
+        %% Extracting the vertices information of the triangles
+
+        sizeTriangle=size(triangles);
+        CADOutputTemp = [];
+        for iterateTriangles=1:sizeTriangle(1)
+%             indexCADOutput=countRows;
+            if isfield(s.amf,'Attributes')
+                switch s.amf.Attributes.unit
+                    case 'millimeter'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*0.001;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*0.001;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*0.001;   
+                    case 'inch'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*0.0254;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*0.0254;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*0.0254;   
+                    case 'foot'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*0.3048;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*0.3048;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*0.3048;   
+                    case 'micrometer'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*1e-6;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*1e-6;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*1e-6;   
+                    case 'meter'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*1;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*1;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*1;                   
+                    otherwise
+                        error('xmlreader does not support this unit.');
+                end
```

Comment:
```suggestion
                switch s.amf.Attributes.unit
                    case 'micrometer'
                        unitConversion = 1e-6;
                    case 'millimeter'
                        unitConversion = 1e-3;
                    case 'meter'
                        unitConversion = 1;
                    case 'kilometer'
                        unitConversion = 1e3;
                    case 'inch'
                        unitConversion = 0.0254;
                    case 'foot'
                        unitConversion = 0.3048;
                    case 'mile'
                        unitConversion = 1609.34;
                    otherwise
                        error('xmlreader does not support this unit.');
                end

                v1 = getTriangleVertex(stemp, iterateVolume, iterateTriangles, 'v1', sizeVolume) * unitConversion;
                v2 = getTriangleVertex(stemp, iterateVolume, iterateTriangles, 'v2', sizeVolume) * unitConversion;
                v3 = getTriangleVertex(stemp, iterateVolume, iterateTriangles, 'v3', sizeVolume) * unitConversion;
```